spec/support/service_account_authorization_context.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+RSpec.shared_context 'with service account authentication' do |service_account_id, scopes|
```

Comment:
💡 This will allow us to avoid stubbing out service account authentication by instead using a shared context to set up the service account config and generate an authorization header to send with requests.

To use, just include the authorization header in the headers of your request.

```ruby
get some_api_path, headers: service_account_auth_header
```

And include the shared context which generates all of the required set-up for a fully integrated test and returns the authorization header. Just pass it the service account id and scopes.

```ruby
include_context 'with service account authentication', 'service-name', ['http://www.example.com/path/for/scope']
```

Reply:
nice!